### PR TITLE
Issue 302: Missing partitions + extra logging around exception

### PR DESCRIPTION
**Pull request by chbatey**
This contains @nivox's solution rebased as well as additional logging around wake up exceptions

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @nivox,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by lightbend-cla-validator** [comment]
Hi @nivox,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala [hunk 1]
```diff
@@ -254,11 +267,18 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
 
   def poll(): Unit = {
     val wakeupTask = context.system.scheduler.scheduleOnce(settings.wakeupTimeout) {
+      log.warning("KafkaConsumer poll is taking significantly longer ({}ms) to return from poll then the configured poll interval ({}ms). Waking up consumer to avoid thread starvation. Thread dump:", settings.wakeupTimeout.toMillis, pollTimeout().toMillis)
```
Comment (patriknw) [inline core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: remove the "Thread dumps:" at the end, belongs to next log message

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala [hunk 2]
```diff
@@ -254,11 +267,18 @@ private[kafka] class KafkaConsumerActor[K, V](settings: ConsumerSettings[K, V])
...
+      if (settings.wakeupDebug) {
+        // See what the KafkaConsumer is up to. Should we do just this dispatcher?
```
Comment (patriknw) [inline core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: good to dump as much as possible until we have this issue completely under control

File: core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala [hunk 3]
```diff
@@ -56,12 +59,15 @@ private[kafka] abstract class SingleSourceLogic[K, V, Msg](
     self.watch(consumer)
 
     val partitionAssignedCB = getAsyncCallback[Set[TopicPartition]] { newTps =>
+      // Is info too much here? Will be logged at startup / rebalance
```
Comment (patriknw) [inline core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala]: perhaps use info/debug depending on the wakeupDebug flag

**Comment by nivox** [comment]
Looks good to me too...

I've signed to CLA, so now the pull request should be able to go through.

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala (hunk 2 above)
Comment (chbatey) [reply core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: There's a heart beat thread that'll be useful to know if it is active too, so will leave this in

File: core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala (hunk 3 above)
Comment (chbatey) [reply core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala]: +1

File: core/src/main/scala/akka/kafka/KafkaConsumerActor.scala (hunk 1 above)
Comment (chbatey) [reply core/src/main/scala/akka/kafka/KafkaConsumerActor.scala]: Doh

**Comment by chbatey** [comment]
Thanks @nivox, i'll give others a little more time to review and try and release a new version later today.

**Pull request merged**